test/Sets/Star.jl

```diff
@@ -35,4 +35,12 @@ for N in [Float64, Float32, Rational{Int}]
     @test basis(S) == N[1 0; 0 1]
     @test center(S) == N[0, 0]
     @test predicate(S) == P
+
+    # intersection with HalfSpace
+    H = HalfSpace(N[0, 1], N(0))
+    I = intersection(S, H)
+    intersection!(S, H)
+    addconstraint!(P, H)
+    @test isequivalent(I, P)
+    @test isequivalent(S, P)
```

Comment:
i think the problem is that `P` is also held by `S` (by creation, see [here](https://github.com/JuliaReach/LazySets.jl/blob/master/src/convert.jl#L1200)), hence we need to make a copy